src/com/google/javascript/jscomp/DefaultPassConfig.java

```diff
@@ -184,6 +185,38 @@ void maybeInitializeModuleRewriteState() {
     }
   }
 
+  private char[] getReservedNamingFirstChars(
```

Comment:
Looks good generally. I would move these two methods to CompilerOptions.

```
  char[] getReservedNamingFirstChars() {
    char[] reservedChars = null;
    if (this.polymerVersion != null && this.polymerVersion > 1) {
      reservedChars = PolymerPass.PROPERTY_RESERVED_FIRST_CHARS;
    } else if (this.angularPass) {
      reservedChars = AngularPass.PROPERTY_RESERVED_FIRST_CHARS;
    }
    if (this.anonymousFunctionNaming.getReservedCharacters() != null) {
      reservedChars =
          Chars.concat(this.anonymousFunctionNaming.getReservedCharacters(), reservedChars);
    }
    return reservedChars;
  }
```

Also, it'd be good if we could have an integration test that enables one of these passes and checks that property renaming uses appropriate names, but I'm not sure how to write one that is not overly long.